### ISSUE TITLE
Remove Old Files in ModelCheckpoint

### ISSUE THREAD

**Issue by tpinetz** [fchollet/keras#5404]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

My problem with the current ModelCheckpoint callback is that when I pass in the validation accuracy as a parameter in the filename I get flooded with model checkpoints. To combat this I would propose a flag in the ModelCheckpoint called `keep_only_last_file`  that checks if the old save should be deleted. In combination with the `save_best_only` flag only the best model is kept.

I would propose following API: `keras.callbacks.ModelCheckpoint(filepath, monitor='val_loss', verbose=0, save_best_only=False, save_weights_only=False, keep_only_last_file=False, mode='auto', period=1)`. I am targeting everyone who wants to use the formatting options in the ModelCheckpoint without saving multiple models.

I can implement this myself.

**Comment by Zlash92** [comment]
Agreed

**Comment by felix-hilden** [comment]
This can be achieved with using a file name with no wildcards for epoch or loss. It's a suboptimal solution for the time being, but does work.

**Comment by Zlash92** [comment]
True, but the formatting options are pretty nice to have.

**Comment by liyxi** [comment]
And people may want to save more than only one checkpoints. I suppose we can have a feature just like `keep_checkpoint_max` in tf estimator:

```python
tf.estimator.RunConfig(model_dir=model_dir,
                       keep_checkpoint_max=3)
```

**Comment by stefan-falk** [comment]
This really seems like an issue for a lot of people e.g.

- https://github.com/tensorflow/tensorflow/issues/30695
- https://github.com/tensorflow/tensorflow/issues/33763

and of course this one.

I wonder, why is there no simple argument such as `keep_checkpoint_max` for the `ModelCheckpoint`?
